src/main/java/xyz/gvital/todo/mapper/TaskHistoryMapper.java

```diff
@@ -0,0 +1,15 @@
+package xyz.gvital.todo.mapper;
+
+import org.mapstruct.Mapper;
+import xyz.gvital.todo.config.MapperConfig;
+import xyz.gvital.todo.dto.TaskHistoryResponseDTO;
+import xyz.gvital.todo.model.TaskHistory;
+
+import java.util.List;
+
+@Mapper(config = MapperConfig.class)
+public interface TaskHistoryMapper {
+    TaskHistoryResponseDTO toDto(TaskHistory model);
+
+    List<TaskHistoryResponseDTO> toDto(List<TaskHistory> model);
```

Comment:
This approach is more common:
```suggestion
    TaskHistoryResponseDTO toDto(TaskHistory model);
```

Reply:
the idea was to declare two overloaded mappers. and let Java pick proper one. So, a little bit simplifying my app code